## Problem Statement:

A factory produces four different products, Diapers, Sanitary Pads, Facial Tissues and Toilet Papers and that their daily produced number of Diapers,  Sanitary Pads, Facial Tissues and Toilet papers is $x_{1}$, $x_{2}$, $x_{3}$ and $x_{4}$ respectively. Based on the following conditions, the factory management wants to come up with a production schedule that would maximize profit. 

## Conditions:

- The profit per unit of product is \\$10, \\$11, \\$8, and \\$12.5 for Diapers, Sanitary Pads, Facial Tissues and Toilet Papers respectively

- Due to manpower constraints, the total number of units produced per day can’t exceed 45

- The raw materials required for producing the aforementioned products are wood pulp and fiber. For producing a single unit of Diaper, 5 units of wood pulp is consumed. Each unit of Sanitary Pad requires 7 units of wood pulp and 2 units of fiber. Each unit of Facial Tissues needs 2 units of wood pulp and 1 unit of fiber. Finally, each unit of Toilet paper requires 3 units of fiber

- Due to the transportation and storage constraints, the factory can consume up to 100 units of wood pulp and 75 units of fiber per day

As the problem is related to production optimization, the environment can be safely assumed to be a controlled one with little to no uncertainty. Hence, we can go ahead with a mathematical model as a solution. 
For the above given problem statement and associated conditions, the mathematical model can be defined like this:

Maximize: 10$x_{1}$ + 11$x_{2}$ + 8$x_{3}$ + 12.5$x_{4}$ (Profit)

## Constraints:

$x_{1}$ + $x_{2}$ + $x_{3}$ + $x_{4}$ $\leq$ 45 (Manpower)

5$x_{1}$ + 7$x_{2}$ + 2$x_{3}$ $\leq$ 100 (Wood pulp)

2$x_{2}$ + 1$x_{3}$ + 3$x_{4}$  $\leq$ 75 (Fiber)

$x_{1}$, $x_{2}$, $x_{3}$, $x_{4}$  $\geq$ 0

## Solution:

We would be using a general-purpose and open-source Linear Programming modeling package in python named PuLP. It is a highly suitable option for a diverse set of optimization problems due to its exceptional qualities such as user-friendliness, versatility, compatibility with other Python libraries, continual support from a thriving community, and open-source availability. 

## Step 1: Importing the PuLP library and defining the mathematical model object

In [1]:
#!pip install --upgrade pip
#!pip install pulp

from pulp import LpProblem, LpMaximize, LpVariable, LpStatus, lpSum

In [2]:
opt_model = LpProblem(name="production-planning", sense=LpMaximize)

Here the model object is defined using the LpProblem module with sense as LpMaximize as we want to maximize the profit made from the production.

## Step 2: Defining the decision variables 

In [3]:
x = {i: LpVariable(name=f"x{i}", lowBound=0) for i in range(1, 5)}

In [4]:
print(x)

{1: x1, 2: x2, 3: x3, 4: x4}


## Step 3: Adding Constraints

In [5]:
opt_model += (lpSum(x.values()) <= 45, "Manpower")
opt_model += (5 * x[1] + 7 * x[2] + 2 * x[3] <= 100, "Wood Pulp")
opt_model += (2 * x[2] + 1 * x[3] + 3 * x[4] <= 75, "Fiber")

## Step 4: Setting the Objective Function

In [6]:
opt_model += (10 * x[1] + 11 * x[2] + 8 * x[3] + 12.5 * x[4], "Objective Function")

## Step 5: Utilizing the default solver method for solving the given optimization problem

In [7]:
status = opt_model.solve()

## Step 6: Get the Results

In [8]:
print(f"status: {opt_model.status}, {LpStatus[opt_model.status]}")
print(f"objective: {opt_model.objective.value()}")

for var in x.values():
    print(f"{var.name}: {var.value()}")

for name, constraint in opt_model.constraints.items():
    print(f"{name}: {constraint.value()}")

status: 1, Optimal
objective: 512.5
x1: 20.0
x2: 0.0
x3: 0.0
x4: 25.0
Manpower: 0.0
Wood_Pulp: 0.0
Fiber: 0.0


**From the above output, we can see that the model suggests producing 20 units of $x_{1}$ i.e. Diapers and 25 units of $x_{4}$ i.e. Toilet Papers for maximum profit ($512.5) under the given constraints.**